toxcore/forwarding.h

```diff
@@ -0,0 +1,108 @@
+/* SPDX-License-Identifier: GPL-3.0-or-later
+ * Copyright © 2019-2022 The TokTok team.
+ */
+
+#ifndef C_TOXCORE_TOXCORE_FORWARDING_H
+#define C_TOXCORE_TOXCORE_FORWARDING_H
+
+#include "DHT.h"
+#include "network.h"
+
+#define SENDBACK_IPPORT 0
+#define SENDBACK_FORWARD 1
+#define SENDBACK_TCP 2
+
+#define MAX_SENDBACK_SIZE (0xff - 1)
+#define MAX_FORWARD_DATA_SIZE (MAX_UDP_PACKET_SIZE - (1 + 1 + MAX_SENDBACK_SIZE))
+
+#define MAX_FORWARD_CHAIN_LENGTH 4
+
+#define MAX_PACKED_IPPORT_SIZE (1 + SIZE_IP6 + sizeof(uint16_t))
+
+typedef struct Forwarding Forwarding;
+
+non_null()
+DHT *forwarding_get_dht(Forwarding *forwarding);
+
+/**
+ * Send data to forwarder for forwarding via chain of dht nodes.
+ * Destination is last key in the chain.
+ * Maximum data_length is MAX_FORWARD_DATA_SIZE.
+ * Maximum chain_length MAX_FORWARD_CHAIN_LENGTH, minimum 1.
```

Comment:
`chain_keys` is undocumented. Can you make it doxygen-style comments with `@param` and stuff? That way this becomes more obvious. It's important to document that `chain_keys` is `key size * chain_length` in size.

Also, this function needs to check its inputs. E.g. check that chain_length is actually `<= MAX_FORWARD_CHAIN_LENGTH` and data_length etc.

Reply:
done